api/server.go

```diff
@@ -178,6 +185,7 @@ func (s *Server) build(ctx context.Context) {
 			},
 		})
 
+		router.Use(s.customMiddlewares...)
```

Comment:
Not sure if there's better place to inject custom middlewares, for example, another option is to put it right after the line
```
ginEngine := gin.Default()
ginEngine.Use(s.customMiddlewares...)
```

Since the original code has ajaxHeadersMiddleware and errorLogMiddleware, I picked a location close to that.